pkgs/development/tools/rust/cargo-modules/default.nix

```diff
@@ -0,0 +1,23 @@
+{ lib, rustPlatform, fetchFromGitHub }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "cargo-modules";
+  version = "0.5.0";
+
+  src = fetchFromGitHub {
+    owner = "regexident";
+    repo = pname;
+    rev = version;
+    sha256 = "0y6ag8nar85l2fh2ca41fglkzc74cv1p5szxrhk1jdqnd2qzhvjp";
+  };
+
+  cargoSha256 = "0m5r36p57w4vw2g3hg12s38ay328swjb0qfl381xwb2xqx10g8kx";
+
+  meta = with lib; {
+    description = "A cargo plugin for showing a tree-like overview of a crate's modules";
+    homepage = "https://github.com/regexident/cargo-modules";
+    license = with licenses; [ mpl20 ];
+    maintainers = with maintainers; [ rvarago ];
+    platforms = [ "x86_64-linux" ];
```

Comment:
```suggestion
```
set by rustPlatform, and this package should also work on any other arch/OS, right?

Reply:
Oh, true! I oversaw it. About working on other arch/OS, I suppose it should, yes. But given that I only have a Linux/x86-64 box to test, I'd opted-in for enforcing the platform.

I've lifted the restriction as suggested.

Thanks ^^.

Reply:
platforms is more about what hydra is going to build than what the maintainer supports. If we know something is really not working and upstream does not support it we can restrict it.